Code/GraphMol/Canon.cpp

```diff
@@ -1255,5 +1274,71 @@ void canonicalizeFragment(ROMol &mol, int atomIdx,
     std::cerr<<"----------------------------------------->"<<std::endl;
 #endif
 }
+void rankMolAtoms(const ROMol &mol, std::vector<unsigned int> &res,
+                  bool breakTies = true, bool includeChirality = true,
+                  bool includeIsotopes = true);
+void canonicalizeEnhancedStereo(ROMol &mol,
+                                const std::vector<unsigned int> *atomRanks) {
+  const auto &sgs = mol.getStereoGroups();
+  if (sgs.empty()) {
+    return;
+  }
+
+  std::vector<unsigned int> lranks;
+  if (!atomRanks) {
+    bool breakTies = true;
+    rankMolAtoms(mol, lranks, breakTies);
+    atomRanks = &lranks;
```

Comment:
Since `lranks` is defined locally and will be destroyed when leaving the function, `atomRanks` will end up returning an invalid address.

Reply:
I think this is ok since `atomRanks` is passed by value (so the value in the caller won't be modified)?

Reply:
:+1: